interpreter/cling/include/cling/Interpreter/Value.h

```diff
@@ -40,18 +98,19 @@ namespace cling {
   /// and parameters for function calls.
   ///
   class CLING_LIB_EXPORT Value {
-  protected:
+  public:
     ///\brief Multi-purpose storage.
     ///
     union Storage {
-      long long m_LL;
-      unsigned long long m_ULL;
+#define X(type, name) type m_##name;
```

Comment:
Adding more storage types seems unnecessary. Anything integer should be stored as `int64_t` or `uint64_t` depending whether it's signed or not, and then cast upon reading, to keep `Value` simple.

Reply:
That allows us to keep track of the precise format and helps us do the concrete conversions now in `getAs`. The idea is to provide a preprocessor generated mapping between the underlying clang type and its compile-time notion. See next comment.